x-pack/legacy/plugins/ml/server/routes/system.ts

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { Request } from 'hapi';
+import { RequestHandlerContext } from 'kibana/server';
+import { wrapError } from '../client/error_wrapper';
+import { mlLog } from '../client/log';
+import { privilegesProvider } from '../lib/check_privileges';
+import { isSecurityDisabled } from '../lib/security_utils';
+import { spacesUtilsProvider } from '../lib/spaces_utils';
+import { licensePreRoutingFactory } from '../new_platform/licence_check_pre_routing_factory';
+import { RouteInitialization } from '../new_platform/plugin';
+
+/**
+ * System routes
+ */
+export function systemRoutes({
+  router,
+  xpackMainPlugin,
+  spacesPlugin,
+  cloud,
+}: RouteInitialization) {
+  async function getNodeCount(context: RequestHandlerContext) {
+    const filterPath = 'nodes.*.attributes';
+    const resp = await context.ml!.mlClient.callAsInternalUser('nodes.info', {
```

Comment:
This call to `nodes.info` was originally being done with `callWithInternalUser` from `callWithInternalUserFactory` - this should be replaced by the NP equivalent available on `context` -> `context.core.elasticsearch.adminClient.callAsInternalUser`